app/assets/src/components/visualizations/table/BaseTable.jsx

```diff
@@ -29,21 +38,28 @@ class BaseTable extends React.Component {
   constructor(props) {
     super(props);
 
-    // TOOD: move this to componentDidUpdate and remove from state?
     this.state = {
       activeColumns: this.props.initialActiveColumns,
       columns: this.setDefaults(this.props.columns),
     };
   }
 
+  // Need to update the columns immediately, otherwise there will be a render
+  // where the data has been updated but the columns haven't.
+  static getDerivedStateFromProps(props, state) {
+    if (props.columns !== state.prevPropsColumns) {
+      return {
+        activeColumns: props.initialActiveColumns,
+        columns: addDefaultsToColumns(props.columns, props.defaultColumnWidth),
+        prevPropsColumns: props.columns,
+      };
+    }
+    return null;
+  }
+
   setDefaults = columns => {
```

Comment:
Makes not sense to define a whole different function when you can just redefined the signature of this one.

Reply:
Done. Renamed to `setColumnDefaults` for clarity.

Reply:
Actually when I ran it, it failed, and now I remember why I did this.

You can't call `this.setDefaults` in a static function (`getDerivedStateFromProps`)

Reply:
yeah, i meant to make it static. I still like within the class though...

Reply:
Done.